sycl/CMakeLists.txt

```diff
@@ -95,34 +108,108 @@ add_custom_target(sycl-headers ALL
 COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/CL ${dst_dir}/CL
 COMMENT "Copying SYCL headers ...")
 
-# Configure SYCL headers
-install(DIRECTORY "${sycl_inc_dir}/." DESTINATION "${LLVM_INST_INC_DIRECTORY}" COMPONENT sycl-headers)
-
-# Configure OpenCL header and ICD loader
-include_directories(AFTER "${sycl_inc_dir}" "${OpenCL_INCLUDE_DIRS}")
-link_libraries(${OpenCL_LIBRARIES})
-
 # SYCL runtime library
 add_subdirectory(source)
 
 # SYCL toolchain builds all components: compiler, libraries, headers, etc.
+set(SYCL_TOOLCHAIN_DEPS
+  sycl
+  clang
+  clang-offload-wrapper
+  clang-offload-bundler
+  llc
+  llvm-as
+  llvm-ar
+  llvm-dis
+  llvm-no-spir-kernel
+  llvm-spirv
+  llvm-link
+  llvm-objcopy
+  opt)
 add_custom_target( sycl-toolchain
-  DEPENDS sycl
-          clang
-          clang-offload-wrapper
-          clang-offload-bundler
-          llc
-          llvm-as
-          llvm-ar
-          llvm-dis
-          llvm-no-spir-kernel
-          llvm-spirv
-          llvm-link
-          llvm-objcopy
-          opt
+  DEPENDS ${SYCL_TOOLCHAIN_DEPS}
   COMMENT "Building SYCL compiler toolchain..."
 )
 
 add_subdirectory( test )
 add_subdirectory( unittests )
 add_subdirectory( tools )
+
+# Configure SYCL toolchain installation targets
+add_custom_target(deploy-sycl-toolchain)
+
+set(SYCL_BUNDLE_DEPS ${SYCL_TOOLCHAIN_DEPS} opencl clang-resource-headers)
+
+foreach(comp ${SYCL_BUNDLE_DEPS})
```

Comment:
This code relies on assumption that target names do match names of associated components.